docs/ibc/1_IBC_ARCHITECTURE.md

```diff
@@ -0,0 +1,141 @@
+# 1: IBC Architecture
+
+> This is an overview of the high-level architecture & dataflow of the IBC protocol.
+
+> For a broad set of protocol design principles, see [here](./2_IBC_DESIGN_PRINCIPLES.md).
+
+> For definitions of terms used in IBC specifications, see [here](./3_IBC_TERMINOLOGY.md).
+
+This document outlines the architecture of the authentication, transport, and ordering layers of the IBC protocol stack. This document does not describe specific protocol details — those are contained in individual ICSs.
+
+## What is IBC?
+
+The *inter-blockchain communication protocol* is designed for use as a reliable, authenticated module-to-module communication protocol between modules running on independent distributed ledgers across an untrusted network layer.
+
+## What is IBC not?
+
+IBC is not (only) an atomic-swap protocol: arbitrary cross-chain data transfer and computation is supported.
+
+IBC is not (only) a token transfer protocol: token transfer is a possible application-layer use of the IBC protocol.
+
+IBC is not (only) a sharding protocol: there is no single state machine being split across chains, but rather a diverse set of different state machines on different chains which share some common interfaces.
+
+IBC is not (only) a layer-two scaling protocol: all chains implementing IBC exist on the same "layer", although they may occupy different points in the network topology, and there is no single root chain or single validator set.
+
+## Motivation
+
+The two predominant blockchains, Bitcoin and Ethereum, support about seven and about twenty transactions per second respectively. Both have been operating at capacity in recent past despite still being utilized primarily by a userbase of early-adopter enthusiasts, far from the desired mainstream adoption. Throughput limitations are a fundamental limitation of distributed state machines, since every (validating) node in the network must process every transaction and store all state. Faster consensus algorithms, such as [Tendermint](https://github.com/tendermint/tendermint), may increase throughput by a large constant factor but will be unable to scale indefinitely for this reason. In order to support the transaction throughput, application diversity, and cost efficiency required to facilitate wide deployment of distributed ledger applications, execution and storage must be split across many independent consensus instances which can run concurrently.
```

Comment:
Several comments on this first paragraph:

- two predominant -> two of the most predominant
- support -> currently support
- "far from desired mainstream adoption" -> I would reformulate, because for Bitcoin some argue it's not the case. A lot of people in the Bitcoin community do not see throughput as a fundamental limitation for bitcoin's use case (store of value). I would say that throughput is a limitation for most blockchain use cases, but not necessarily all of them. 
- "since every (validating) node in the network must process every transaction and store all state" -> is this really the limiting factor, or is it that nodes must agree on sets of transactions sequentially? I would say it's the latter

Reply:
- they are predominant by market capitalization, noted that
- updated
- Satoshi's paper refers to "peer-to-peer digital cash", and it is that use case which I am referring to here; but the point is not to argue with Bitcoin adherents, just to explain the problem. What do you suggest? I just cut the last part of the sentence for now.
- It's the limiting factor on the number of transactions in a single block (or round of consensus), yes. What do you mean by "nodes must agree on sets of transactions sequentially" - that they must run consensus algorithms?

Reply:
> What do you suggest?

Just say that "throughput is a limitation for most blockchain use cases"

> What do you mean by "nodes must agree on sets of transactions sequentially" - that they must run consensus algorithms?

I'm saying that without IBC we're essentially in a purely sequential world. IBC enables concurrent designs. A validator can scale their ressources to have more storage/bandwidth, but on a single ledger the validator set will always be limited because of the fact that there is a single global state that must be updated all at once.

Reply:
> Just say that "throughput is a limitation for most blockchain use cases"

Updated.

> I'm saying that without IBC we're essentially in a purely sequential world. IBC enables concurrent designs. A validator can scale their ressources to have more storage/bandwidth, but on a single ledger the validator set will always be limited because of the fact that there is a single global state that must be updated all at once.

I would agree with all of this - does something in the text contradict it?

The last sentence in this paragraph is:

```
In order to support the transaction throughput, application diversity, and cost efficiency required to facilitate wide deployment of distributed ledger applications, execution and storage must be split across many independent consensus instances which can run concurrently
```

Reply:
The last sentence is great!

I guess the confusing sentence was "since every (validating) node in the network must process every transaction and store all state". To me it sounded like the limitation was storage/bandwidth, and not the fact that the state must be updated globally. 

In a concurrent model nodes would still have to store all state, but subsets of the state would be update-able concurrently. Maybe rephrasing the sentence slightly would help clarify

Reply:
Added "... and communicate with other validating nodes."